test/Tests/src/Data/Text_Spec.enso

```diff
@@ -204,6 +206,61 @@ spec =
             'a'.each builder2.append
             builder2.to_vector . should_equal ['a']
 
+        Test.specify "should check for contains using Unicode normalization" <|
+            "Hello".contains "ell" . should_be_true
+
+            "Cześć".contains 's\u{301}' . should_be_true
+            "Cześć".contains 'c\u{301}' . should_be_true
+            "Cześć".contains 'ść' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ś' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ć' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ść' . should_be_true
+            "Cześć".contains 'sc' . should_be_false
+            'Czes\u{301}c\u{301}'.contains 'sc' . should_be_false
+
+            ## TODO what do we do with that?? Since the standard decomposition
+               splits 'ś' into 's+{accent}', 'ś'.contains 's', but I don't think
+               this is the expected behaviour...
+            "Cześć".contains 's' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 's' . should_be_true
```

Comment:
I've added an analogous test and there it works as expected (i.e. `ś` does not contain `s` in regex matching mode). That's an argument for making this work correctly in exact matching mode - we want to be consistent.
Also that's reassuring, because if Regex didn't support this properly and we still wanted that property - it could have been very hard to 'fix' the Regex implementation.

Reply:
One solution to that could be to just write the contains serach manually, using the BreakIterator, ensuring that it looks at whole grapheme clusters correctly. This is likely going to be slightly slower than what we have now (ICU `Normalizer2` preprocess step + Java `contains`), but may be the only way to go to retain correctness.

...

After a long and deep dive into ICU4J API, I've found `StringSearch` which should do what we need in an efficient manner. Will try it out.

Interestingly it allows setting locale - I have no idea how changing locale can influence the search in general - but noting this as it may be something we may want to explore (although I'd set up a separate chore task for it instead of digging into it right now - but that's up to discussion).